2048_fib_mini project/moves_handling_functions.c

```diff
@@ -0,0 +1,192 @@
+//moves_handling_functions.c
+#include<stdio.h>
+#include"fibonacci_checkers.c"
+//functions declaration
+
+void downMove(int *,int );
+void shiftdown(int *,int , int , int ,int);
+void upMove(int *,int );
+void shiftUp(int *,int , int , int ,int);
+void leftMove(int *,int );
+void shiftLeft(int *,int , int , int ,int);
+void rightMove(int *,int );
+void shiftRight(int *,int , int , int ,int);
+int handleInputs(char [],int ,int ,int *);
+void removeZeroesOfColumnsForDownMove(int *,int );
+void removeZeroesOfColumnsForUpMove(int *,int );
+void removeZeroesOfRowsForRightMove(int *,int );
+void removeZeroesOfRowsForLeftMove(int *,int );
+
+int handleInputs(char name[],int moves,int n,int *a){
+	char c;
+	printf("enter w - UP / s - DOWN / d - RIGHT / a - LEFT\n\t\t\t any key to exit \n\t\t\tgame is saved automatically Don't worry \n: ");
+	scanf(" %c",&c);
+	if (c == 'w')
+		upMove(a,n);
+	else if (c == 's')
+		downMove(a,n);
+	else if (c == 'd')
+		rightMove(a,n);
+	else if (c == 'a')
+		leftMove(a,n);
+	else return 1;
+	return 0;
+}
+
+void removeZeroesOfColumnsForDownMove(int *a,int n){ 
+	int j=n-1;
+	for(int k=0;k<n;k++){ //this gives a coloumn number in every iteration
+		j=n-1;
+		for(int i=n-1;i>=0;i--){ //this is for row index
+			if(*(a+i*n+k)!=0){
+				*(a+j*n+k)=*(a+i*n+k);
+				j--;
+			}
+		}
+		while(j>=0){
+			*(a+j*n+k)=0;
+			j--;
+		}
+	}
+}
+
+void removeZeroesOfColumnsForUpMove(int *a,int n){ 
+	int j=0;
+	for(int k=0;k<n;k++){ //this gives a coloumn number in every iteration
+		j=0;
+		for(int i=0;i<n;i++){ //this is for row index
+			if(*(a+i*n+k)!=0){
+				*(a+j*n+k)=*(a+i*n+k);
+				j++;
+			}
+		}
+		while(j<n){
+			*(a+j*n+k)=0;
+			j++;
+		}
+	}
+}
+
+void removeZeroesOfaRowLeftMove(int *a,int n){
+	int j=0;
+	for(int k=0;k<n;k++){		//this gives a coloumn number in every iteration
+		j=0;
+		for(int i=0;i<n;i++){ //this is for row 
+			if(*(a+k*n+i)!=0){
+				*(a+k*n+j)=*(a+k*n+i);
```

Comment:
You can use a pointer as an array. Say `int *a` is a pointer `*(a+i)` is as same as `a[i]`. That would be clear, neat, clean and easy to understand. What you did is absolutely correct but not preferred.

Reply:
mine is a pointer to 2D array and to pass 2D array as arg we need to specify atleast one size but here size is  n (size entered by user). so i can't send 2D array pointer so i should use pointer type notation for sure.

Reply:
You can pass `**ptr`, if I'm not wrong. Please try with that and see if that works for you. If yes come up with changes else just put a comment here and resolve the conversation.